scribus/ui/propertywidget_opentypefontfeatures.cpp

```diff
@@ -39,20 +39,153 @@ void PropertyWidget_OpenTypeFontFeatures::changeEvent(QEvent *e)
 	QWidget::changeEvent(e);
 }
 
+void PropertyWidget_OpenTypeFontFeatures::languageChange()
+{
+		CommonCheck->setChecked(true);
+		NormalCaRadio->setChecked(true);
+		NormalRadio->setChecked(true);
+		DefaultStyleRadio->setChecked(true);
+		ContextualCheck->setChecked(true);
+		DiscretinoryCheck->setChecked(false);
+		HistoricalCheck->setChecked(false);
+		SubscriptRadio->setChecked(false);
+		SuperscriptRaido->setChecked(false);
+		SmallRadio->setChecked(false);
+		SmallFromCRadio->setChecked(false);
+		PetiteRadio->setChecked(false);
+		PetiteCapRadio->setChecked(false);
+		UnicaseRadio->setChecked(false);
+		TiltingRadio->setChecked(false);
+		LininRadio->setChecked(false);
+		OldStyleRadio->setChecked(false);
+		ProporionalRadio->setChecked(false);
+		TabularRadio->setChecked(false);
+		DiagonalRadio->setChecked(false);
+		StackedRadio->setChecked(false);
+		OrdinalCheck->setChecked(false);
+		SlashedZeroCheck->setChecked(false);
+		groupBox->setToolTip(tr("Capitals"));
+		groupBox_2->setToolTip(tr("Numbers"));
+		groupBox_3->setToolTip(tr("Ligatures"));
+		groupBox_4->setToolTip(tr("Position"));
+}
+
 void PropertyWidget_OpenTypeFontFeatures::showFontFeatures(QString s)
 {
 	if (!m_ScMW || m_ScMW->scriptIsRunning())
 		return;
-	fontfeatures->setText(s);
+	QStringList fontFeatures = s.split(',');
+	for (int i = 0; i < fontFeatures.count(); i++)
+	{
+		if (fontFeatures[i] == "+clig")
+			ContextualCheck->setChecked(true);
+		else if (fontFeatures[i] == "-clig")
+			ContextualCheck->setChecked(false);
+		else if (fontFeatures[i] == "+liga")
+			CommonCheck->setChecked(true);
+		else if (fontFeatures[i] == "-liga")
```

Comment:
You should set `ContextualCheck->setChecked(true)` and `CommonCheck->setChecked(true)` before the loop, and set them unchecked only of there is `-clig` and `-liga`.